azure-servicebus/src/test/java/com/microsoft/azure/servicebus/ProxySelectorTests.java

```diff
@@ -0,0 +1,56 @@
+package com.microsoft.azure.servicebus;
+
+import com.microsoft.azure.servicebus.primitives.ConnectionStringBuilder;
+import com.microsoft.azure.servicebus.primitives.ServiceBusException;
+import com.microsoft.azure.servicebus.primitives.TransportType;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.net.*;
+import java.time.Duration;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+public class ProxySelectorTests {
+
+
+    @Test
+    public void proxySelectorConnectFailedInvokeTest() throws Exception
+    {
+        // set up proxy selector with a bad address in order to check that the connectFailed() method is invoked
+        int noProxyPort = 8888;
+        final CompletableFuture<Void> connectFailedTask = new CompletableFuture<>();
+
+        ProxySelector.setDefault(new ProxySelector() {
+            @Override
+            public List<Proxy> select(URI uri) {
+                List<Proxy> proxies = new LinkedList<>();
+                proxies.add(new Proxy(Proxy.Type.HTTP, new InetSocketAddress("localhost", noProxyPort)));
+                return proxies;
+            }
+
+            @Override
+            public void connectFailed(URI uri, SocketAddress sa, IOException ioe) {
+                connectFailedTask.complete(null);
+            }
+        });
+
+        ConnectionStringBuilder connectionStringBuilder = new ConnectionStringBuilder(TestUtils.getNamespaceConnectionString());
+        connectionStringBuilder.setTransportType(TransportType.AMQP_WEB_SOCKETS);
+        connectionStringBuilder.setOperationTimeout(Duration.ofSeconds(10));
+
+        try {
+            QueueClient sendClient = new QueueClient(connectionStringBuilder, ReceiveMode.PEEKLOCK);
+        } catch (ServiceBusException ex) {
+            // todo? Better assert statement?
+            Assert.assertEquals(
+               "Error{condition=amqp:connection:framing-error, description='connection aborted', info=null}",
+                ex.getLocalizedMessage());
+        }
+
+        connectFailedTask.get(2, TimeUnit.SECONDS);
+    }
```

Comment:
This test is not required. My approach to testing WebSockets + Proxy would be just running all the tests by changing connection string transport type + proxy environment variables. If needed, we can run all the tests once with TCP transport type, once with WebSocket transport type and once with WebSockets + Proxy. We don't need these tests that are testing too much of the specific internals.

Reply:
Those would be positive test cases. This is a negative test case to make sure we are actually going through proxy. Lets say there's a bug in the code and it is not even touching proxy and bypassing it; all the test cases would successfully pass and we wouldn't be able to catch it.
IMO, lets keep this test.

---
In reply to: [221411288](https://github.com/Azure/azure-service-bus-java/pull/285#discussion_r221411288) [](ancestors = 221411288)